### DISCUSSION TITLE
Access page context in links function

### DISCUSSION THREAD

**Discussion by QuentinWidlocher** [remix-run/remix#1258] [Q&A] [answered]
Hi everyone, I see we can get the loader data in the `meta` function to use our page context in our meta tags (perfect for a title for example)

Why can't we do the same in the `links` function ?
In my case, I'd wish to preload an external url stored in an object I fetch in my loader.

**Comment by ryanflorence** [comment]
We used to have it but removed it, the reasoning is found in the 0.19 release notes under "Removal of data in links({ data })": https://github.com/remix-run/remix/releases/tag/v0.19.0#:~:text=Removal%20of%20data%20in%20links(%7B%20data%20%7D)

**Comment by sergiodxa** [comment, marked as answer]
You can use the useMatches hook to add those `<link rel="preload" />` in the <head> of your document.

Use the useMatches to access the data of each route loader, and get the resources you need, then render the link tags.

**Reply by QuentinWidlocher** [reply]
Thanks, adding a `link` tag is what I've done, and it matches Ryan's answer too 👌
